### DISCUSSION TITLE
Is there a repository de facto processing time limitation?

### DISCUSSION THREAD

**Discussion by JoeJennings** [renovatebot/renovate#18721] [Request Help] [answered]
Renovate OnPrem -> v 2.5.2, Self-hosted, against GitHub enterprise.

Renovate is taking a long time (76 minutes) to process a repository, which is an issue that can be resolved; however it does create a scenario that leads to this question. Is there a repository de facto processing time limitation for pull request creation?  Asking because here's the scenario I'm seeing in the logs, Renovate processes the repository and calls GitHub (enterprise 3.4.1) to get the repository's PR information (...pulls? per_page...) to determine if it should create a PR.  This call fails with a 401 Unauthorized, Bad Credentials, my suspicion is the GitHub token that Renovate is trying to use has expired and thus the reason for the PR data call (pulls API) failure.  Renovate doesn't create a PR, but it does log out that the Repository finished and the duration.

```
{"name":"renovate","hostname":"bfae5f3e0878","pid":6,"level":30,"logContext":"4n3UPVgrA","repository":"test-org/narrative-component-reactor","durationMs":4612977,"msg":"Repository finished","time":"2022-11-02T14:55:26.023Z","v":0}
```

I thought I read Github tokens can only be valid for 10 minutes, is this the de facto processing time limitation for PR creation? 

When I make updates either to the repository or to Renovate configuration so that Renovate is able to process the repository in less than 10 minutes the GitHub "pulls" API is successful and Renovate creates a PR.

Does/could Renovate have logic to retrieve a valid token if processing time is lengthy?  Or is something else going on and my suspicion of the token expiration incorrect?

Another observation is in this scenario this repository either stays in the Renovate job queue or is added back into the Renovate job queue where it again processes for a long time and this pattern continues indefinitely of Renovate processing the repository repeatedly but never creating a PR.

Additional logs:

```
{"name":"renovate","hostname":"bfae5f3e0878","pid":6,"level":20,"logContext":"4n3UPVgrA","repository":"test-org/narrative-component-reactor","token":"***********","err":{"name":"HTTPError","code":"ERR_NON_2XX_3XX_RESPONSE","timings":{"start":1667400925707,"socket":1667400925708,"lookup":1667400925708,"connect":1667400925708,"secureConnect":1667400925709,"upload":1667400925709,"response":1667400925734,"end":1667400925734,"phases":{"wait":1,"dns":0,"tcp":0,"tls":1,"request":0,"firstByte":25,"download":0,"total":27}},"message":"Response code 401 (Unauthorized)","stack":"HTTPError: Response code 401 (Unauthorized)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","options":{"headers":{"user-agent":"WhiteSource Renovate","accept":"application/vnd.github.machine-man-preview+json","authorization":"***********","accept-encoding":"gzip, deflate, br"},"url":"https://githubdev.cerner.com/api/v3/repos/test-org/narrative-component-reactor/pulls?per_page=100&state=all&sort=updated&direction=desc&page=1","hostType":"github","username":"","password":"","method":"GET","http2":false},"response":{"statusCode":401,"statusMessage":"Unauthorized","body":{"message":"Bad credentials","documentation_url":"https://docs.github.com/enterprise/3.4/rest"},"headers":{"server":"GitHub.com","date":"Wed, 02 Nov 2022 14:55:25 GMT","content-type":"application/json; charset=utf-8","content-length":"95","x-github-enterprise-version":"3.4.1","x-github-media-type":"github.v3; param=machine-man-preview; format=json","x-ratelimit-limit":"2000","x-ratelimit-remaining":"1999","x-ratelimit-reset":"1667404525","x-ratelimit-used":"1","x-ratelimit-resource":"core","access-control-expose-headers":"ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset","access-control-allow-origin":"*","x-github-request-id":"dea6bd6d-d797-44f5-bb94-54d4ffc3becd","strict-transport-security":"max-age=31536000; includeSubdomains","x-frame-options":"deny","x-content-type-options":"nosniff","x-xss-protection":"0","referrer-policy":"origin-when-cross-origin, strict-origin-when-cross-origin","content-security-policy":"default-src 'none'","x-runtime-rack":"0.019120","connection":"close"},"httpVersion":"1.1"}},"msg":"GitHub failure: Bad credentials","time":"2022-11-02T14:55:25.735Z","v":0}
```

**Comment by rarkins** [comment, marked as answer]
GitHub app tokens time out after 1 hour

**Reply by JoeJennings** [reply]
Thank you that answers that I believe.
